ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/CSJ.hs

```diff
@@ -22,55 +20,67 @@ import           Test.Consensus.PeerSimulator.Run (SchedulerConfig (..),
+-- trunk of the block tree) with CSJ enabled. What we expect to observe is that
+-- one of the honest peers becomes the dynamo while the others become jumpers.
+-- Because the jumpers will agree to all the jumps, the whole syncing should
+-- happen with CSJ.
 --
--- The final property is that headers should only ever be downloaded once and
--- only from one peer (the dynamo). This is true except when almost caught-up:
--- when the dynamo is caught-up, it gets disengaged and one of the jumpers takes
--- its place and starts serving headers. This might lead to duplication of
--- headers, but only in a window of @jumpSize@ slots near the tip of the chain.
+-- There are two variants of this test: the “happy path” variant features no
+-- adversaries. As such, everything should happen with one dynamo and no
+-- objector. Another variant adds adversaries, so we expect to see some
+-- dynamo-vs-objector action.
 --
--- The boolean differentiates between “synchronous” and “asynchronous”
--- scenarios. In a synchronous scenario, all the honest peers have the same
--- schedule: they serve the chain exactly in the same way. In the asynchronous
--- scenario, a random schedule is generated for each peer (but they still serve
--- the same chain).
-prop_happyPath :: Bool -> Property
-prop_happyPath synchronized =
+-- Regardless, the final property is that “honest” headers should only ever be
+-- downloaded at most once from honest peers. They may however be downloaded
+-- several times from adversaries. This is true except when almost caught-up:
+-- when the dynamo or objector is caught-up, it gets disengaged and one of the
+-- jumpers takes its place and starts serving headers. This might lead to
+-- duplication of headers, but only in a window of @jumpSize@ slots near the tip
+-- of the chain.
+--
+-- The first boolean differentiates between the “happy path” variant and the
+-- variant with adversaries; the second boolean differentiates between
+-- “synchronous” and “asynchronous” scenarios. In a synchronous scenario, all
+-- the honest peers have the same schedule: they serve the chain exactly in the
+-- same way. In the asynchronous scenario, a random schedule is generated for
+-- each peer (but they still serve the same chain).
+prop_CSJ :: Bool -> Bool -> Property
+prop_CSJ happy synchronized =
   forAllGenesisTest
-    ( do
-        gt <- genChains $ pure 0
-        honest <- genHonestSchedule gt
-        numOthers <- choose (1, 3)
-        otherHonests <- if synchronized
-          then pure $ replicate numOthers honest
-          else replicateM numOthers (genHonestSchedule gt)
-        pure $ gt $> mkPeers honest otherHonests
+    ( if synchronized
+        then genChainsWithExtraHonestPeers (choose (2, 4)) (if happy then pure 0 else choose (2, 4))
+          `enrichedWith` genUniformSchedulePoints
+        else genChains (if happy then pure 0 else choose (2, 4))
+          `enrichedWith` genDuplicatedHonestSchedule
```

Comment:
It looks to me like the if condition is inverted here. Synchronized peers should use the same honest schedule according to the documentation.